tests/gateway.ts

```diff
@@ -4131,4 +4131,524 @@ describe("Gateway", () => {
       expect(err.message).to.include("SignerIsNotAuthority");
     }
   });
+
+  describe("PDA Migration and Extension", () => {
+    it("Should migrate PDA to new format with realloc", async () => {
+      // Verify PDA exists and get initial state
+      const pdaBefore = await gatewayProgram.account.pda.fetch(pdaAccount);
+      const accountInfoBefore = await conn.getAccountInfo(pdaAccount);
+      const oldSize = accountInfoBefore.data.length;
+
+      // Call migrate_pda instruction
+      await gatewayProgram.methods
+        .migratePda()
+        .accounts({
+          pda: pdaAccount,
+          payer: wallet.publicKey,
+          systemProgram: anchor.web3.SystemProgram.programId,
+        })
+        .rpc();
+
+      // Verify PDA now has new fields
+      const pdaAfter = await gatewayProgram.account.pda.fetch(pdaAccount);
+      expect(pdaAfter.bump).to.be.a("number");
+      expect(pdaAfter.bump).to.be.gte(0);
+      expect(pdaAfter.bump).to.be.lte(255);
+      expect(pdaAfter.nominatedAuthority).to.be.null;
+
+      // Verify account size increased (or is already at new size if already migrated)
+      const accountInfoAfter = await conn.getAccountInfo(pdaAccount);
+      expect(accountInfoAfter.data.length).to.be.gte(oldSize);
+      // If already migrated, size should be at least the new size
+      if (oldSize >= accountInfoAfter.data.length) {
+        // Already migrated, just verify fields exist
+        expect(accountInfoAfter.data.length).to.be.gte(112); // New size
+      } else {
+        expect(accountInfoAfter.data.length).to.be.gt(oldSize);
+      }
+
+      // Verify existing fields are preserved
+      expect(pdaAfter.nonce.toString()).to.equal(pdaBefore.nonce.toString());
+      expect(Array.from(pdaAfter.tssAddress)).to.be.deep.eq(
+        Array.from(pdaBefore.tssAddress)
+      );
+      expect(pdaAfter.authority.toString()).to.equal(
+        pdaBefore.authority.toString()
+      );
+      expect(pdaAfter.chainId.toString()).to.equal(
+        pdaBefore.chainId.toString()
+      );
+      expect(pdaAfter.depositPaused).to.equal(pdaBefore.depositPaused);
+    });
+
+    it("Should be idempotent if migration called twice", async () => {
+      // Get current state
+      const pdaBefore = await gatewayProgram.account.pda.fetch(pdaAccount);
+      const bumpBefore = pdaBefore.bump;
+
+      // Call migration again
+      await gatewayProgram.methods
+        .migratePda()
+        .accounts({
+          pda: pdaAccount,
+          payer: wallet.publicKey,
+          systemProgram: anchor.web3.SystemProgram.programId,
+        })
+        .rpc();
+
+      // Verify state is unchanged
+      const pdaAfter = await gatewayProgram.account.pda.fetch(pdaAccount);
+      expect(pdaAfter.bump).to.equal(bumpBefore);
+      expect(pdaAfter.nominatedAuthority).to.be.null;
```

Comment:
please also add test to check if only authority can migrate pda, since that is also missing in gateway function